server/config/config.go

```diff
@@ -112,23 +126,30 @@ func (conf *Config) Update(stored StoredConfig, mmconf *model.Config, license *m
 			host = "127.0.0.1"
 		}
 
-		localURL = "http://" + host + ":" + port + mattermostURL.Path
+		localURL = "http://" + host + ":" + port + u.Path
 	}
 
 	conf.StoredConfig = stored
 
-	conf.MattermostSiteURL = *mattermostSiteURL
-	conf.MattermostSiteHostname = mattermostURL.Hostname()
+	conf.MattermostSiteURL = u.String()
 	conf.MattermostLocalURL = localURL
 	conf.PluginURLPath = "/plugins/" + conf.PluginManifest.Id
-	conf.PluginURL = strings.TrimRight(*mattermostSiteURL, "/") + conf.PluginURLPath
+	conf.PluginURL = strings.TrimRight(u.String(), "/") + conf.PluginURLPath
 
 	conf.MaxWebhookSize = 75 * 1024 * 1024 // 75Mb
 	if mmconf.FileSettings.MaxFileSize != nil {
 		conf.MaxWebhookSize = *mmconf.FileSettings.MaxFileSize
 	}
 
 	conf.DeveloperMode = pluginapi.IsConfiguredForDevelopment(mmconf)
+	if developerModeDomains.MatchString(u.Hostname()) {
+		conf.DeveloperMode = true
+	}
+
+	conf.AllowHTTPApps = !conf.MattermostCloudMode || conf.DeveloperMode
```

Comment:
Why should customers on a cloud server not be able to install HTTP apps?

Reply:
This is an @aaronrothschild question, not a technical one, correct?

Reply:
@hanzei @levb I believe the blocker for this is the ongoing effort for oauth scopes. I'm 0/5 on if we should make that block this though